### PR TITLE
CORE-1814: added the ability to log in and store cached copies of acc…

### PR DESCRIPTION
**Pull request by slr71**
…ess tokens

### CODE REVIEW

**Comment by slr71** [comment]
I'm going to merge this now. Reviews are still welcome and appreciated.

File: terrain.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+def terrain_auth_file(environment):
+    """
+    Returns the Terrain auth token file path for the given environment.
+    """
+    return "{0}/.terrain-{1}".format(os.environ["HOME"], environment)
+
+def get_cached_access_token(environment):
+    """
+    Attempts to obtain the access token from $HOME/.terrain-auth-{environment}.
```
Comment (psarando) [inline terrain.py:26]: It looks like `terrain_auth_file` will look for the file at `$HOME/.terrain-{environment}`?
```suggestion
    Attempts to obtain the access token from $HOME/.terrain-{environment}.
```

File: terrain.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    token = None
+    while token is None:
+        username = get_username()
+        password = get_password()
+        token = get_auth_token(environment, username, password)
+    return token
+
+def cache_token(environment, token):
+    """
+    Stores a copy of the access token in $HOME/.terrain-auth-{environment}.
```
Comment (psarando) [inline terrain.py:84]:
```suggestion
    Stores a copy of the access token in $HOME/.terrain-{environment}.
```

File: terrain.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    """
+    auth_file = terrain_auth_file(environment)
+    with open(auth_file, "w") as f:
+        print(token, file=f)
+    os.chmod(auth_file, stat.S_IRUSR | stat.S_IWUSR)
+
+def get_access_token(environment):
+    """
+    Gets the access token to use for Terrain. The access token will be cached in $HOME/.terrain-auth-{environment}. If
```
Comment (psarando) [inline terrain.py:93]:
```suggestion
    Gets the access token to use for Terrain. The access token will be cached in $HOME/.terrain-{environment}. If
```

**Review by psarando — APPROVED** [review]
LGTM 👍

File: terrain.py (hunk 1 above)
Comment (slr71) [reply terrain.py:26]: Ah! Thanks for catching these 👍

**Comment by slr71** [comment]
Thanks for the review. I'll fix the comments in the next PR.

**Pull request merged**